ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -451,3 +472,70 @@ func copyFile(src, dst string) error {
 	}
 	return os.Chmod(dst, srcinfo.Mode())
 }
+
+// findImageRefByArch finds the appropriate image reference in a multi-arch manifest image based on the current platform's OS and processor architecture
+func findImageRefByArch(ctx context.Context, imageRef string, pullSecret []byte, osToFind string, archToFind string) (manifestImageRef string, err error) {
+	manifestList, err := registryclient.GetManifest(ctx, imageRef, pullSecret)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve manifest from image ref, %s: %w", imageRef, err)
+	}
+
+	_, payload, err := manifestList.Payload()
+	if err != nil {
+		return "", fmt.Errorf("failed to get manifest payload: %w", err)
+	}
+
+	m := new(manifestlist.DeserializedManifestList)
```

Comment:
we should name as deserializedManifestList consistently with below

Reply:
Agree. It's been updated.